recipes/renode-cores/helpers/renode_build_with_cmake.ps1

```diff
@@ -0,0 +1,68 @@
+$ErrorActionPreference = "Stop"
+
+$cpuCount = (Get-CimInstance Win32_Processor).NumberOfLogicalProcessors
+
+$env:PATH = "${env:BUILD_PREFIX}/Library/mingw-w64/bin;${env:BUILD_PREFIX}/Library/bin;${env:PREFIX}/Library/bin;${env:PREFIX}/bin;${env:PATH}"
+$env:SRC_DIR = $env:SRC_DIR
+
+$CMAKE = (Get-Command cmake).Source
+
+$CONFIGURATION = "Release"
+$HOST_ARCH = "i386"
+$CMAKE_COMMON = ""
+
+$CORES_PATH = Join-Path -Path $env:SRC_DIR -ChildPath "src\Infrastructure\src\Emulator\Cores"
+
+Push-Location "$env:SRC_DIR\tools\building"
+    bash .\check_weak_implementations.sh
+Pop-Location
+
+$CORES_BUILD_PATH = Join-Path -Path $CORES_PATH -ChildPath "obj\$CONFIGURATION"
+$CORES_BIN_PATH = Join-Path -Path $CORES_PATH -ChildPath "bin\$CONFIGURATION"
+
+$CMAKE_GEN = "-GNinja"
+
+$CORES = @("arm.le", "arm.be", "arm64.le", "arm-m.le", "arm-m.be", "ppc.le", "ppc.be", "ppc64.le", "ppc64.be", "i386.le", "x86_64.le", "riscv.le", "riscv64.le", "sparc.le", "sparc.be", "xtensa.le")
+
+function Update-CMakeLists {
+    param (
+        [string]$filePath,
+        [string]$flagToRemove,
+        [string]$flagToAdd
+    )
+    (gc $filePath) -replace $flagToRemove,$flagToAdd | sc $filePath
+}
+
+Update-CMakeLists "$CORES_PATH\tlib\CMakeLists.txt" "-fPIC" "-Wno-unused-function -Wno-maybe-uninitialized"
+Update-CMakeLists "$CORES_PATH\tlib\tcg\CMakeLists.txt" "-fPIC" "-Wno-unused-function -Wno-maybe-uninitialized"
```

Comment:
It would be much simpler to just add `-Wno-unused-function -Wno-maybe-uninitialized` to the `CXXFLAGS` environment variable and then just use the replace function directly to remove `-fPIC` instead of writing a custom function like this.

Also please provide a comment explaining why `-fPIC` should be removed - there are very few cases where we wouldn't position independent code.

Reply:
I seem to remember that this -fPIC is redundant and creates an extra notification. Also, these CMakeLists.txt may not use the CXXFLAGS, but I should double-check